File: java/program/Laptop.java
// Class Laptop inheriting from Perangkat
public class Laptop extends Perangkat {
    private String cpu;
    private String gpu;
    private int ram; // in GB
    private int storage; // in GB

    // Constructor without parameters
    public Laptop() {
        super();
        this.cpu = "";
        this.gpu = "";
        this.ram = 0;
        this.storage = 0;
    }

    // Constructor with parameters
    public Laptop(String id, String nama, int harga, String merek, String kode_sku, int garansi, String cpu, String gpu, int ram, int storage) {
        super(id, nama, harga, merek, kode_sku, garansi);
        this.cpu = cpu;
        this.gpu = gpu;
        this.ram = ram;
        this.storage = storage;
    }

    // Getter & Setter for cpu
    public String getCpu() {return cpu;}
    public void setCpu(String cpu) {this.cpu = cpu;}

    // Getter & Setter for gpu
    public String getGpu() {return gpu;}
    public void setGpu(String gpu) {this.gpu = gpu;}

    // Getter & Setter for ram
    public int getRam() {return ram;}
    public void setRam(int ram) {this.ram = ram;}

    // Getter & Setter for storage
    public int getStorage() {return storage;}
    public void setStorage(int storage) {this.storage = storage;}
}

File: python/program/Laptop.py
from Perangkat import Perangkat

### Inheritance from Perangkat class
class Laptop(Perangkat):
    ## Constructor
    def __init__(self, id="", nama="", harga=0, merek="", kode_sku="", garansi=0, cpu="", gpu="", ram=0, storage=0):
        super().__init__(id, nama, harga, merek, kode_sku, garansi)
        self.cpu = cpu
        self.gpu = gpu
        self.ram = ram
        self.storage = storage

    ## Setter and getter for cpu
    def set_cpu(self, cpu): self.cpu = cpu  
    def get_cpu(self): return self.cpu

    ## Setter and getter for gpu
    def set_gpu(self, gpu): self.gpu = gpu
    def get_gpu(self): return self.gpu

    ## Setter and getter for ram
    def set_ram(self, ram): self.ram = ram
    def get_ram(self): return self.ram

    ## Setter and getter for storage
    def set_storage(self, storage): self.storage = storage
    def get_storage(self): return self.storage



